src/hotspot/share/classfile/systemDictionary.cpp

```diff
@@ -1121,23 +1101,16 @@ InstanceKlass* SystemDictionary::resolve_from_stream(Symbol* class_name,
                                                      ClassFileStream* st,
                                                      TRAPS) {
 
+  assert(st != NULL, "invariant");
```

Comment:
Is this necessary? We normally don't have asserts for non-null-ness.

Reply:
I moved it from below.  Probably not necessary. The callers pass the address of a heap variable and testing this for NULL really doesn't make any sense.